kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/arbitrary/DateTest.kt

```diff
@@ -212,4 +214,46 @@ class DateTest : WordSpec({
          checkAll(10) { _: Period -> /* No use. Won't reach here if unsupported */ }
       }
    }
+
+   "Arb.yearMonth(minYearMonth, maxYe)" should {
+      "generate valid YearMonths (no exceptions)" {
+         Arb.yearMonth().take(10_000).toList()
+      }
+
+      "generate YearMonths between minYear and maxYear" {
+         val years = mutableSetOf<Int>()
+         val months = mutableSetOf<Int>()
+
+         checkAll(10_000, Arb.yearMonth(YearMonth.of(1998, 1), YearMonth.of(1999, 12))) {
+            years += it.year
+            months += it.monthValue
+         }
+
+         years shouldBe setOf(1998, 1999)
+         months shouldBe (1..12).toSet()
+      }
+
+      "Generate YearMonth always in the range" {
+         repeat(1_000) {
+            val min = YearMonth.of(1970, 1).plusMonths(it.toLong())
+            val max = min.plusYears(20)
+            Arb.yearMonth(min, max).forAll { it >= min && it <= max }
+         }
+      }
```

Comment:
This adds to my previous comment about improving the tests.
I think having some glanularity for months should be validated in the tests as well